src/plugins/application_config/server/opensearch_config_client.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { IScopedClusterClient, Logger } from '../../../../src/core/server';
+
+import { ConfigurationClient } from './types';
+import { ERROR_FOR_EMPTY_INPUT, ERROR_MESSSAGE_FOR_EMPTY_INPUT, isEmpty } from './string_utils';
+
+export class OpenSearchConfigurationClient implements ConfigurationClient {
+  private client: IScopedClusterClient;
+  private configurationIndexName: string;
+  private readonly logger: Logger;
+
+  constructor(
+    openSearchClient: IScopedClusterClient,
+    configurationIndexName: string,
+    logger: Logger
+  ) {
+    this.client = openSearchClient;
+    this.configurationIndexName = configurationIndexName;
+    this.logger = logger;
+  }
+
+  async getEntityConfig(entity: string) {
+    if (isEmpty(entity)) {
+      this.logger.error(ERROR_MESSSAGE_FOR_EMPTY_INPUT);
+      throw ERROR_FOR_EMPTY_INPUT;
+    }
```

Comment:
nit: Can we create separate function for validation, maybe in separate file which only consist validation logic?

Reply:
Moved to a separate function. Indeed looks cleaner!